db/config.cc

```diff
@@ -540,6 +540,8 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     /* Note: does not exist on the listing page other than in above comment, wtf? */
     , commitlog_sync_batch_window_in_ms(this, "commitlog_sync_batch_window_in_ms", value_status::Used, 10000,
         "Controls how long the system waits for other writes before performing a sync in ``batch`` mode.")
+    , commitlog_max_data_lifetime_in_seconds(this, "commitlog_max_data_lifetime_in_seconds", value_status::Used, 24*60*60,
+        "Controls how long data remains in commit log before the system tries to evict it to sstable, regardless of usage pressure. (0 disables)")
```

Comment:
Should be liveupdatable, so if it's needed in production we don't need a rolling restart.

Reply:
No commitlog options are live updatable, since the two configs are separated and the CL does not have a "re-apply config" today. 
Adding that is a different PR.

Reply:
Let's not add new problems if we can.

There is updateable_value, but #7316 is problematic. If we create the config object on the shard it's used in, then we can just use it.

Reply:
Updatable db::config does not help if there is a secondary config + subsystem like CL that does not accept configuration after the fact. While I guess one could just add a "cl->update_max_age(x)" to CL and hook in an observer on the db::config value, the current way db::config -> cl::config looks we'd be breaking the abstraction layers a bit. Maybe we don't care...

Reply:
Added a runtime update of the value.